core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -601,7 +640,14 @@ private IndexingStrategy planIndexingAsNonPrimary(Index index) throws IOExceptio
             // unlike the primary, replicas don't really care to about creation status of documents
             // this allows to ignore the case where a document was found in the live version maps in
             // a delete state and return false for the created flag in favor of code simplicity
-            final OpVsLuceneDocStatus opVsLucene = compareOpToLuceneDocBasedOnVersions(index);
+            final OpVsLuceneDocStatus opVsLucene;
+            if (index.seqNo() != SequenceNumbersService.UNASSIGNED_SEQ_NO) {
```

Comment:
I left an extra comment on the assertion in the else clause:

```
                // This can happen if the primary is still on an old node and send traffic without seq# or we recover from translog
                // created by an old version.
                assert config().getIndexSettings().getIndexVersionCreated().before(Version.V_6_0_0_alpha1_UNRELEASED) :
                    "index is newly created but op has no sequence numbers. op: " + index;
```